### PR TITLE
[Aon timer] Testutils refactor do return status_t

### PR DESCRIPTION
**Pull request by engdoreis**
See https://github.com/lowRISC/opentitan/issues/17495

Fix https://github.com/lowRISC/opentitan/issues/17526

### CODE REVIEW

**Review by pamaury — APPROVED** [review]

File: sw/device/lib/testing/aon_timer_testutils.h [hunk 1]
```diff
@@ -7,38 +7,51 @@
...
  */
-uint32_t aon_timer_testutils_get_us_from_aon_cycles(uint64_t cycles);
+status_t aon_timer_testutils_get_us_from_aon_cycles(uint64_t cycles,
+                                                    uint32_t *microseconds);
 
 /**
  * Configure wakeup counter for a number of AON clock cycles.
+ *
  * @param cycles The number of AON clock cycles.
+ * @return The result of the operation.
```
Comment (timothytrippel) [inline sw/device/lib/testing/aon_timer_testutils.h:35]: aon_timer pointer param missing

File: sw/device/lib/testing/aon_timer_testutils.h [hunk 2]
```diff
@@ -7,38 +7,51 @@
...
  * @param bark_cycles The number of AON clock cycles till barking.
  * @param bite_cycles The number of AON clock cycles till biting.
  * @param pause_in_sleep Don't increment while sleeping.
+ * @return The result of the operation.
  */
-void aon_timer_testutils_watchdog_config(const dif_aon_timer_t *aon_timer,
-                                         uint32_t bark_cycles,
-                                         uint32_t bite_cycles,
-                                         bool pause_in_sleep);
+status_t aon_timer_testutils_watchdog_config(const dif_aon_timer_t *aon_timer,
```
Comment (timothytrippel) [inline sw/device/lib/testing/aon_timer_testutils.h:51]: aon_timer pointer param missing

File: sw/device/lib/testing/aon_timer_testutils.h [hunk 3]
```diff
@@ -47,7 +60,9 @@ void aon_timer_testutils_watchdog_config(const dif_aon_timer_t *aon_timer,
  * the design stops generating events that may affect the simulation from
  * terminating cleanly. The wakeup and watchdog timer controls are read back
  * to ensure the written value successfully crossed the AON clock domain.
+ *
+ * @return The result of the operation.
  */
-void aon_timer_testutils_shutdown(const dif_aon_timer_t *aon_timer);
+status_t aon_timer_testutils_shutdown(const dif_aon_timer_t *aon_timer);
```
Comment (timothytrippel) [inline sw/device/lib/testing/aon_timer_testutils.h:66]: aon_timer pointer param missing

**Review by timothytrippel — APPROVED** [review]

File: sw/device/lib/testing/aon_timer_testutils.h (hunk 1 above)
Comment (engdoreis) [reply sw/device/lib/testing/aon_timer_testutils.h:35]: Fixed

File: sw/device/lib/testing/aon_timer_testutils.h (hunk 3 above)
Comment (engdoreis) [reply sw/device/lib/testing/aon_timer_testutils.h:66]: Fixed.

File: sw/device/lib/testing/aon_timer_testutils.h [hunk 4]
```diff
@@ -7,38 +7,51 @@
...
 #include <stdint.h>
 
+#include "sw/device/lib/base/status.h"
 #include "sw/device/lib/dif/dif_aon_timer.h"
 
 /**
- * Returns the number of AON cycles corresponding to the given microseconds.
+ * Compute the number of AON cycles corresponding to the given microseconds.
+ *
+ * @param[out] cycles The number of AON clock cycles.
```
[the remark this answers is not in the document]
Comment (engdoreis) [reply sw/device/lib/testing/aon_timer_testutils.h:16]: Fixed.

**Pull request merged**